tests/sources/test_mysql.py

```diff
@@ -666,6 +676,214 @@ async def test_validate_config_when_host_empty_then_raise_error():
+            [],
+            {},
+            [],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: one custom query
+            [TABLE_ONE],
+            [
+                {
+                    "tables": [TABLE_ONE],
+                    "query": "SELECT * FROM *",
+                    "id_columns": [ID_ONE],
+                }
+            ],
+            [ID_ONE],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # valid: two custom queries
+            [TABLE_ONE, TABLE_TWO],
+            [
+                {
+                    "tables": [TABLE_ONE],
+                    "query": "SELECT * FROM *",
+                    "id_columns": [ID_ONE],
+                },
+                {
+                    "tables": [TABLE_ONE, TABLE_TWO],
+                    "query": "SELECT * FROM *",
+                    "id_columns": [ID_ONE, ID_TWO],
+                },
+            ],
+            [ID_ONE, ID_TWO],
+            SyncRuleValidationResult.valid_result(
+                SyncRuleValidationResult.ADVANCED_RULES
+            ),
+        ),
+        (
+            # invalid: additional property present
+            [TABLE_ONE],
+            [ID_ONE],
+            [
+                {
+                    "tables": [TABLE_ONE],
+                    "query": "SELECT * FROM *",
+                    "id_columns": [ID_ONE],
+                    "additional_property": True,
+                }
+            ],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+        (
+            # invalid: tables field missing
+            [TABLE_ONE],
+            [ID_ONE],
+            [{"query": "SELECT * FROM *", "id_columns": [ID_ONE]}],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+        (
+            # invalid: query field missing
+            [TABLE_ONE],
+            [ID_ONE],
+            [{"tables": [TABLE_ONE], "id_columns": [ID_ONE]}],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
+        ),
+        (
+            # invalid: id_columns field missing
+            [TABLE_ONE],
+            [ID_ONE],
+            [{"tables": [TABLE_ONE], "query": "SELECT * FROM *"}],
+            SyncRuleValidationResult(
+                SyncRuleValidationResult.ADVANCED_RULES,
+                is_valid=False,
+                validation_message=ANY,
+            ),
```

Comment:
This means this is a backwards incompatible change, right? I don't think that's ok. `id_columns` needs to be optional.

Reply:
Yes, this field is optional and this will work without `id_columns`. it fails because of wrong arguments. I have updated it.